go-controller/pkg/node/gateway_init_linux_test.go

```diff
@@ -350,143 +504,70 @@ var _ = Describe("Gateway Init Operations", func() {
-				brNextHopIP   string = "169.254.33.1"
-				brNextHopCIDR string = brNextHopIP + "/24"
-				systemID      string = "cb9ec8fa-b409-4ef3-9f42-d9283c47aac6"
-				nodeSubnet    string = "10.1.1.0/24"
-			)
-
-			fexec.AddFakeCmdsNoOutputNoError([]string{
-				"ovs-vsctl --timeout=15 --may-exist add-br br-local",
-			})
-			fexec.AddFakeCmd(&ovntest.ExpectedCmd{
-				Cmd:    "ovs-vsctl --timeout=15 --if-exists get interface br-local mac_in_use",
-				Output: brLocalnetMAC,
-			})
-			fexec.AddFakeCmdsNoOutputNoError([]string{
-				"ovs-vsctl --timeout=15 set bridge br-local other-config:hwaddr=" + brLocalnetMAC,
-			})
-			fexec.AddFakeCmd(&ovntest.ExpectedCmd{
-				Cmd:    "ovs-vsctl --timeout=15 --if-exists get Open_vSwitch . external_ids:ovn-bridge-mappings",
-				Output: "",
-			})
-			fexec.AddFakeCmdsNoOutputNoError([]string{
-				"ovs-vsctl --timeout=15 set Open_vSwitch . external_ids:ovn-bridge-mappings=" + util.PhysicalNetworkName + ":br-local",
-				"ovs-vsctl --timeout=15 --if-exists del-port br-local " + legacyLocalnetGatewayNextHopPort +
-					" -- --may-exist add-port br-local " + localnetGatewayNextHopPort + " -- set interface " + localnetGatewayNextHopPort + " type=internal mtu_request=" + mtu + " mac=00\\:00\\:a9\\:fe\\:21\\:01",
-			})
-			fexec.AddFakeCmd(&ovntest.ExpectedCmd{
-				Cmd:    "ovs-vsctl --timeout=15 --if-exists get Open_vSwitch . external_ids:system-id",
-				Output: systemID,
-			})
-			fexec.AddFakeCmd(&ovntest.ExpectedCmd{
-				Cmd: "ip rule",
-				Output: "0:	from all lookup local\n32766:	from all lookup main\n32767:	from all lookup default\n",
-			})
-			fexec.AddFakeCmdsNoOutputNoError([]string{
-				"ip rule add from all table " + localnetGatewayExternalIDTable,
-			})
-			fexec.AddFakeCmdsNoOutputNoError([]string{
-				"ip route list table " + localnetGatewayExternalIDTable,
-			})
+	Context("for localnet operations", func() {
+		const (
+			v4BrNextHopIP       string = "169.254.33.1"
+			v4BrNextHopCIDR            = v4BrNextHopIP + "/24"
+			v4NodeSubnet        string = "10.1.1.0/24"
+			v4localnetGatewayIP string = "169.254.33.2"
+
+			v6BrNextHopIP       string = "fd99::1"
+			v6BrNextHopCIDR            = v6BrNextHopIP + "/64"
+			v6NodeSubnet        string = "2001:db8:abcd:0012::0/64"
+			v6localnetGatewayIP string = "fd99::2"
+		)
+		var (
+			brNextHopCIDRs       []*netlink.Addr
+			ipts                 []*util.FakeIPTables
+			v4ipt, v6ipt         util.IPTablesHelper
+			expectedIpTableRules []map[string]util.FakeTable
+		)
+		BeforeEach(func() {
+			brNextHopCIDRs = make([]*netlink.Addr, 0)
+			ipts = make([]*util.FakeIPTables, 0)
+			expectedIpTableRules = make([]map[string]util.FakeTable, 0)
+			v4ipt, v6ipt = util.SetFakeIPTablesHelpers()
+		})
 
-			existingNode := v1.Node{ObjectMeta: metav1.ObjectMeta{
-				Name: nodeName,
-			}}
+		It("sets up a IPv4 localnet gateway", func() {
+			nextHopCIDRIPv4, err := netlink.ParseAddr(v4BrNextHopCIDR)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
you can use `ovntest.MustParseIPNet()` here instead
(likewise in the other tests)

Reply:
I looked at this but the return values are different. Hence I did not make this change.

Reply:
oh, sorry, I was thinking of `netlink.ParseIPNet()`, not `netlink.ParseAddr()`